lib/project_types/script/layers/application/push_script.rb

```diff
@@ -9,19 +9,11 @@ def call(ctx:, force:)
             script_project = ScriptProject.current
             task_runner = Infrastructure::TaskRunner.for(ctx, script_project.language, script_project.script_name)
             ProjectDependencies.install(ctx: ctx, task_runner: task_runner)
-            BuildScript.call(
-              ctx: ctx,
-              task_runner: task_runner,
-              extension_point_type: script_project.extension_point_type,
-              script_name: script_project.script_name,
-              description: script_project.description
-            )
+            BuildScript.call(ctx: ctx, task_runner: task_runner, script_project: script_project)
```

Comment:
From what I see, the `ScriptProject` does not belong to any layers.  Should it?  I see that it also calls `Layers::Domain::ExtensionPointAssemblyScriptSDK.language`.  I find this a bit messy.  It feels like we should understand better `ScriptProject`'s relationship to `Script` and `metadata` and etc.  Any thoughts on how to improve its abstraction?

Reply:
I agree, and I do have some thoughts on that! Up until now, `ScriptProject` has been a convenience class to subclass `ShopifyCli::Project`. I think instead we could convert `ScriptProject` into an explicitly defined entity and use `ShopifyCli::Project` as an implementation detail in the repository. The cross-layer access could then be abstracted into an application service. This way, we would be able to better model our relationships to other entities and follow the existing patterns we have already created. Since you bring up how the current implementation feels messy, maybe it's time to think that through. What do you think?

Reply:
Yeah.  I don't remember if we still need `ScriptProject` to inherit from `ShopifyCli::Project`.  I do feel it's a good time to rethink its relationships.  I think for now, we can punt on it into a separate tech debt ticket.  I am good with all the other changes here.  👍